gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java

```diff
@@ -250,10 +254,37 @@ private static void configureErrorProneOptions(
                         check);
                 return false;
             }
-            return true;
+            return filterOutPreconditions.test(check);
         });
     }
 
+    /** Filters out preconditions checks if the required libraries are not on the classpath. */
+    public static Predicate<String> filterOutPreconditions(FileCollection compileClasspath) {
+        boolean hasSafeLogging = !compileClasspath.filter(file -> file.getName().startsWith("safe-logging-")).isEmpty();
+        boolean hasPreconditions =
+                // The real 'preconditions' brings in 'safe-logging'. Because of inaccurate jar name checks, we
+                // use that fact to ensure we're not picking up another jar named 'preconditions' by mistake.
+                hasSafeLogging
+                        && !compileClasspath.filter(file -> file.getName().startsWith("preconditions-")).isEmpty();
+        return check -> {
+            if (!hasPreconditions && check.equals("PreferSafeLoggingPreconditions")) {
+                log.info(
+                        "Disabling check PreferSafeLoggingPreconditions as 'com.palantir.safe-logging:safe-logging' "
+                                + "missing from source set for {}",
+                        compileClasspath);
+                return false;
+            }
+            if (!hasSafeLogging && check.equals("PreferSafeLoggableExceptions")) {
```

Comment:
These live in the preconditions library, we only need the `hasPreconditions` check

Reply:
ah fair, but i'd still like to check for both since I don't trust trust just the check on a jar starting with `preconditions-`